.github/actions/xahau-configure-ccache/action.yml

```diff
@@ -14,18 +14,53 @@ inputs:
     description: 'How to check compiler for changes'
     required: false
     default: 'content'
+  is_main_branch:
+    description: 'Whether the current branch is the main branch'
+    required: false
+    default: 'false'
+  main_cache_dir:
+    description: 'Path to the main branch cache directory'
+    required: false
+    default: '~/.ccache-main'
+  current_cache_dir:
+    description: 'Path to the current branch cache directory'
+    required: false
+    default: '~/.ccache-current'
 
 runs:
   using: 'composite'
   steps:
     - name: Configure ccache
       shell: bash
       run: |
-        mkdir -p ~/.ccache
-        export CONF_PATH="${CCACHE_CONFIGPATH:-${CCACHE_DIR:-$HOME/.ccache}/ccache.conf}"
+        # Create cache directories
+        mkdir -p ${{ inputs.main_cache_dir }} ${{ inputs.current_cache_dir }}
+        
+        # Set compiler check globally
+        ccache -o compiler_check=${{ inputs.compiler_check }}
+        
+        if [ "${{ inputs.is_main_branch }}" == "true" ]; then
+          # Main branch: use main branch cache
+          ccache --set-config=cache_dir="${{ inputs.main_cache_dir }}"
+          echo "CCACHE_DIR=${{ inputs.main_cache_dir }}" >> $GITHUB_ENV
+          
+          # Configure settings for main cache
+          export CONF_PATH="${{ inputs.main_cache_dir }}/ccache.conf"
```

Comment:
Is it necessary to change the location of the conf file?
You only need to change the cache path and secondary path referenced in the same conf file.